apps/pollbook/barcode-scanner-daemon/src/unitech_ts100_daemon.rs

```diff
@@ -0,0 +1,438 @@
+    let listener = UnixListener::bind(UDS_PATH)?;
+    fs::set_permissions(UDS_PATH, fs::Permissions::from_mode(0o660))?;
+    log!(
+        event_id: EventId::SocketServerBind,
+        message: format!("UDS bound on {UDS_PATH}"),
+        disposition: Disposition::Success
+    );
+
+    Ok(listener)
+}
+
+/// Reads from any `BufRead`, parses AAMVA documents, and writes JSON+"\n" to any Write.
+pub async fn run_read_write_loop(
+    running: &Arc<AtomicBool>,
+    reader: &mut dyn BufRead,
+    clients: Arc<Mutex<Vec<UnixStream>>>,
+) {
+    let mut buf = Vec::with_capacity(MAX_AAMVA_DOCUMENT_SIZE);
+    let mut start = Instant::now();
+
+    while running.load(Ordering::SeqCst) {
+        if start.elapsed() > HEARTBEAT_LOG_INTERVAL {
+            start = Instant::now();
+            log!(EventId::Heartbeat; EventType::SystemStatus);
+        }
+
+        buf.clear();
+        match reader.read_until(TS100_DATA_TERMINATOR, &mut buf) {
+            Ok(_) => {
+                if buf.len() > MAX_AAMVA_DOCUMENT_SIZE {
+                    log!(
+                        EventId::ParseError,
+                        "Scan data size of {} exceeded limit of {}",
+                        buf.len(),
+                        MAX_AAMVA_DOCUMENT_SIZE
+                    );
+                    continue;
+                }
+
+                if buf.ends_with(&[TS100_DATA_TERMINATOR]) {
+                    buf.pop();
+                }
+                if buf.is_empty() {
+                    continue;
+                }
+                // We expect the input sequence to start with the compliance indicator
+                // on its own line so we just ignore it and skip to the next line.
+                // Existence of the compliance indicator is not enforced.
+                if buf.as_slice() == COMPLIANCE_INDICATOR {
+                    continue;
+                }
+
+                // Parse and emit JSON
+                let str = match from_utf8(&buf) {
+                    Ok(str) => str,
+                    Err(e) => {
+                        log!(
+                            event_id: EventId::ParseError,
+                            message: format!("Error parsing read bytes as UTF-8: {e}")
+                        );
+                        continue;
+                    }
+                };
+
+                let doc = match AamvaDocument::from_str(str) {
+                    Ok(doc) => doc,
+                    Err(err @ AamvaParseError::DataTooLong(_, _)) => {
+                        // Exit if data is in AAMVA format but doesn't follow AAMVA spec
+                        log!(
+                            event_id: EventId::ParseError,
+                            message: format!("Unexpected data in AAMVA format. Error was: {err}"),
+                            event_type: EventType::SystemAction,
+                            disposition: Disposition::Failure
+                        );
+                        exit(1);
```

Comment:
@benadida note this hard exit when data matches AAMVA format but exceeds allowed length per spec

Reply:
does the daemon then restart?